vpr/src/route/router_lookahead_extended_map.h

```diff
@@ -0,0 +1,74 @@
+#ifndef EXTENDED_MAP_LOOKAHEAD_H_
+#define EXTENDED_MAP_LOOKAHEAD_H_
+
+#include <vector>
+#include "physical_types.h"
+#include "router_lookahead.h"
+#include "router_lookahead_map_utils.h"
+#include "router_lookahead_cost_map.h"
+#include "vtr_geometry.h"
+
+// Implementation of RouterLookahead based on source segment and destination connection box types
+class ExtendedMapLookahead : public RouterLookahead {
+  private:
+    ///<Look-up table from SOURCE/OPIN to CHANX/CHANY of various types
+    util::t_src_opin_delays src_opin_delays;
+
+    ///<Look-up table from CHANX/CHANY to SINK/IPIN of various types
+    util::t_chan_ipins_delays chan_ipins_delays;
+
+    /**
+     * @brief Returns the cost to get to a input delta_x/y location when starting from a SOURCE or OPIN node
+     * @param from_node starting node
+     * @param delta_x x delta value that is the distance between the current node to the target node
+     * @param delta_y y delta value that is the distance between the current node to the target node
+     * @param criticality_fac criticality of the current connection
```

Comment:
say between 0 (all congestion) and 1 (all timing)

Reply:
Done